python/ml4ir/applications/ranking/model/layers/masking.py

```diff
@@ -181,13 +212,13 @@ def call(self, inputs, mask=None, training=None):
             Feature tensor where mask_rate of records' features have been masked out (or set to 0.)
             Shape: [batch_size, sequence_len, feature_dim]
         """
-        batch_size, sequence_len, feature_dim = tf.shape(inputs)[0], tf.shape(inputs)[1], tf.shape(inputs)[2]
-        if self.mask_at_inference or training:
-            mask = tf.cast(tf.math.greater(tf.random.uniform([batch_size, feature_dim]), self.mask_rate), dtype=tf.float32)
-            mask = tf.expand_dims(mask, axis=1)
-            mask = tf.tile(mask, [1, sequence_len, 1])
-            return tf.multiply(inputs, mask)
+        if training:
+            if QueryFeatureMask.masking_config[MonteCarloInferenceKey.USE_FIXED_MASK_IN_TRAINING]:
+                return self.apply_fixed_mask(inputs, training)
+            else:
+                return self.apply_stochastic_mask(inputs, training)
         else:
-            return inputs
-
-
+            if QueryFeatureMask.masking_config[MonteCarloInferenceKey.USE_FIXED_MASK_IN_TESTING]:
+                return self.apply_fixed_mask(inputs, training)
+            else:
+                return self.apply_stochastic_mask(inputs, training)
```

Comment:
@mohazahran this is much better. We can refactor this further if you want to reduce if-else ladder.

Maybe something like
```
if (training and self.use_fixed_mask_in_training) or ((not training) and self.use_fixed_mask_in_testing)):
    return self.apply_fixed_mask(inputs, training)
else:
    return self.apply_stochastic_mask(inputs, training)
```

Reply:
ping